### ISSUE TITLE
Thread management

### ISSUE THREAD

**Issue by ddurieux** [fusioninventory/fusioninventory-for-glpi#1431]
---

Author Name: **Stoat Wblr** (@Stoatwblr)
Original Redmine Issue: 2549, http://forge.fusioninventory.org/issues/2549
Original Date: 2014-06-15

---

Currently when glpi calls FI-agent, the agent uses _all_ its threads for the requested task

In a small network (or test deployment) this is a problem because there may only be one agent.

It would be better to be able to specify the number of threads allowed for each task.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **David Durieux** (@ddurieux)
Original Date: 2014-10-26T23:16:00Z

---

It can be interesting but threads number is directly related with resources of the computer where agent run, not the task.

So I think it's not a good feature. 

Perhaps you have good arguments ;)

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Stoat Wblr** (@Stoatwblr)
Original Date: 2015-05-17T23:33:13Z

---

Apologies for long delay.

The problem as I see it is that an agent may have significantly more threading capability than the master assigns per task.

The current setup means that an agent operates tasks "serially". ie: it runs N threads for one task and then N threads for another task, but it can only perform one task at a time.

I think it would be better to set an absolute-maximum thread limit in the client configuration and then allow tasks to be executed in parallel, up to ABSMAX threads.
